normandy/recipes/api/serializers.py

```diff
@@ -102,6 +103,43 @@ class Meta:
             'is_approved',
         ]
 
+    def validate_arguments(self, value):
+        # Get the schema associated with the selected action
+        try:
+            schema = Action.objects.get(name=self.initial_data.get('action')).arguments_schema
+        except:
+            raise serializers.ValidationError('Could not find arguments schema.')
+
+        schemaValidator = Validator(schema)
+        errorResponse = {}
+        errors = sorted(schemaValidator.iter_errors(value), key=lambda e: e.path)
+
+        for error in errors:
```

Comment:
It might be a good idea here to add a comment that shows the expected shape of the data here too.